common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1123,7 +1123,9 @@ def _block_matches_all(block_data):
             settings['children'] = qualifiers.pop('children')
         for block_id, value in course.structure['blocks'].iteritems():
             if _block_matches_all(value):
-                items.append(block_id)
+                # Check if the block is in the course tree
+                if self.has_path_to_root(block_id, course):
```

Comment:
Yeah, this was my idea to fix the discussion tab problem too.

About the performance degradation — it may be possible to ballpark it locally but just running get items on a course with a ton of items with and without the change. I'm guessing the degradation wouldn't be that bad, because courses don't have that many items, and this doesn't require any module instantiation to work.

 Also, @mushtaqak , this logic should be kept behind a kwarg for now, since there may be other logic that depends on get_items returning orphans too.